airflow_kjo/kubernetes_job_launcher.py

```diff
@@ -44,9 +40,14 @@ def __init__(
         self.kube_job_client = get_kube_job_client(self.kube_client)
         self.kube_pod_client = get_kube_pod_client(self.kube_client)
         self.sleep_time = 5
-        self.tail_logs_every = tail_logs_every
-        self.tail_logs_line_count = tail_logs_line_count
-        self.tail_logs_only_at_end = tail_logs_only_at_end
+        self.yaml_obj = None
+        self.name = None
+        self.namespace = None
+
+    def __call__(self, yaml_obj):
+        self._validate_job_yaml(yaml_obj)
+        self.yaml_obj = yaml_obj
```

Comment:
When you look at other methods (e.g. _get_name_namespace) and you wonder "where is yaml_obj set". The intuition is that it should be set with __init__ because that's where attributes are initialized. Having separate injection points to define attributes makes it harder to read imho.

Reply:
true